base/compiler/ssair/ir.jl

```diff
@@ -1270,10 +1312,8 @@ function maybe_erase_unused!(extra_worklist, compact, idx, callback = x->nothing
     stmt === nothing && return false
     if compact_exprtype(compact, SSAValue(idx)) === Bottom
         effect_free = false
-    elseif isa(compact.result[idx][:info], MethodResultPure)
```

Comment:
Late comment: seems like this isn't yet being used to set the IR_FLAG_EFFECT_FREE flag? Shouldn't be a major impact in the meantime

Reply:
Here it is: https://github.com/JuliaLang/julia/pull/40066/files#diff-c08d0dffd36a142536203edece93dbe8656a68c607fd06256903d4b2ed8f2942R1284-R1286. We do also have a test for this :), but good catch, would have been easy to miss.

Reply:
Ah good. Inlining also seems like an unexpected place for it, but yeah, I can see why searching for MethodResultPure failed—it was just above the fold.

Reply:
Yes, it's a bit odd, but it does kinda make sense, since inlining is what incorporates our general interprocedural information into the local IR.